tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -295,6 +297,20 @@ TangoErrorType TangoRosNode::OnTangoServiceConnected() {
   return TANGO_SUCCESS;
 }
 
+void TangoRosNode::RequestADFPermissionAndWaitForAnswer() {
+  UpdateAndPublishTangoStatus(TangoStatus::NEED_TO_REQUEST_ADF_PERMISSION);
+  while (tango_status_ != TangoStatus::ADF_PERMISSION_REQUEST_ANSWERED) {
+    ros::Duration(0.1).sleep();
+  }
```

Comment:
I chose to trigger permission request from the C++ side because this is where we know exactly when and which permission we need. For example, if the user reconnect to Tango using the ros command  line tool, the Java side does not know that a service call to `/tango/connect has` been made.
Here we are using the `/tango/status` topic to trigger the permission request from C++ to Java, and know when the request has been answered.